modules/custom/openy_repeat/src/Controller/RepeatController.php

```diff
@@ -136,28 +196,40 @@ public function getLocations() {
    * Get detailed info about Location (aka branch).
    */
   public function getLocationsInfo() {
-    $nids = \Drupal::entityQuery('node')
-      ->condition('type','branch')
-      ->execute();
-    $branches = Node::loadMultiple($nids);
-
-    $data = [];
-    foreach ($branches as $node) {
-      $days = $node->get('field_branch_hours')->getValue();
-      $address = $node->get('field_location_address')->getValue();
-      if (!empty($address[0])) {
-        $address = array_filter($address[0]);
-        $address = implode(', ', $address);
+    $data = $tags = [];
+    $cid = 'openy_repeat:locations_info';
+    if ($cache = $this->cache->get($cid)) {
+      $data = $cache->data;
+    }
+    else {
+      $nids = $this->entityQuery
+        ->get('node')
+        ->condition('type','branch')
+        ->execute();
+      $nids_chunked = array_chunk($nids, 20, TRUE);
+      foreach ($nids_chunked as $chunk) {
+        $branches = $this->entityTypeManager->getStorage('node')->loadMultiple($chunk);
+        if (!empty($branches)) {
+          foreach ($branches as $node) {
+            $days = $node->get('field_branch_hours')->getValue();
+            $address = $node->get('field_location_address')->getValue();
+            if (!empty($address[0])) {
+              $address = array_filter($address[0]);
+              $address = implode(', ', $address);
+            }
+            $data[$node->title->value] = [
+              'nid' => $node->nid->value,
+              'title' => $node->title->value,
+              'email' => $node->field_location_email->value,
+              'phone' => $node->field_location_phone->value,
+              'address' => $address,
+              'days' => !empty($days[0]) ? $this->getFormattedHours($days[0]) : [],
+            ];
+            $tags[] = 'node:' . $node->nid->value;
+          }
+        }
       }
-
-      $data[$node->title->value] = [
-        'nid' => $node->nid->value,
-        'title' => $node->title->value,
-        'email' => $node->field_location_email->value,
-        'phone' => $node->field_location_phone->value,
-        'address' => $address,
-        'days' => !empty($days[0]) ? $this->getFormattedHours($days[0]) : [],
-      ];
+      $this->cache->set($cid, $data, CacheBackendInterface::CACHE_PERMANENT, $tags);
```

Comment:
please add tag 'node_list' so whenever new class being added this cache also will be invalidated